### PR TITLE
make python and pylint more up-to-date

### PR DESCRIPTION
**Pull request by pkasprzyk**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ def main():
         all_deps += extras[group_name]
     extras['all'] = all_deps
     extras['dev'] = [
-        'pylint==2.4.2',
+        'pylint==2.9.6',
         'pylintfileheader==0.0.2',
```
Comment (Raalsky) [inline setup.py:20]: There is `0.3.2` version available. Maybe we should also bump here?

File: setup.py [hunk 2]
```diff
@@ -16,7 +16,7 @@ def main():
         all_deps += extras[group_name]
     extras['all'] = all_deps
     extras['dev'] = [
-        'pylint==2.4.2',
+        'pylint==2.9.6',
```
Comment (Raalsky) [inline setup.py:19]: What do you think about `>=2.9.6`?

File: setup.py (hunk 1 above)
Comment (pkasprzyk) [reply setup.py:20]: good point -- in neptune-client we just have unpinned version, I can do the same here

File: setup.py (hunk 2 above)
Comment (pkasprzyk) [reply setup.py:19]: When I ran `2.13.8` (current) on `neptune-client`, I got a lot of errors (pylint adds/changes validations between versions...) -- so right now I'd like to have a passing build + consistent version, and talk about upgrade later (in separate PR).

**Review by Raalsky — APPROVED** [review]

**Pull request merged**